docs/zephyr_02_functional_requirements.sdoc

```diff
@@ -1317,6 +1317,248 @@ As a Zephyr OS user I want to... ???,
 Architectural & interface REQ
 <<<
 
+[SECTION]
+TITLE: Counting Semaphores
+
+[REQUIREMENT]
+UID: ZEP-127
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Counting Semaphores
+TITLE: Build-time registration for pre-kernel init phase notification
+STATEMENT: >>>
+Zephyr Counting Semaphores shall build-time register with Zephyr System Initialization for notification of running in the Zephyr Pre-Kernel initialization phase.
```

Comment:
**issue:** This formulation introduces a lot of jargon/technical terms which as of today cannot be resolved from within the RQTs spec:
* build-time register
* Zephyr System Initialization
* Zephyr Pre-Kernel initialization phase

IMHO, We need to re-state the problem these constructs have helped us solve.

Reply:
The additional technical terms come from decomposing higher layers of the system. Unfortunately, we started at the implementation instead of the user needs/stakeholder constraints.